src/coingecko_api/base_client.py

```
@@ -0,0 +1,97 @@
+                async with self._session.request(
+                    method, 
+                    url, 
+                    params=params, 
+                    headers=headers,
+                    timeout=aiohttp.ClientTimeout(total=self._config.timeout)
+                ) as response:
+                    response.raise_for_status()
+                    return await response.json()
+
+            except (aiohttp.ClientError, asyncio.TimeoutError) as e:
+                logger.warning(f"Request failed (Attempt {attempt + 1}): {e}")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)